digid_eherkenning/views/digid.py

```diff
@@ -110,9 +117,40 @@ def get(self, request):
         return HttpResponseRedirect(self.get_success_url())
 
 
-class DigiDLogoutView(View):
-    # TODO logout view for SLO
-    pass
+class DigiDLogoutView(LogoutView):
+    """
+    1. local logout from django app
+    2. Single logout with HTTP-redirect
+    """
+
+    @method_decorator(never_cache)
+    def dispatch(self, request, *args, **kwargs):
+        name_id = self.get_name_id(request)
+
+        if not name_id:
```

Comment:
This is for the cases where either a user is logged in but not logged in with DigiD or they are not logged in at all?

I'm wondering if it would be better to raise an error in these cases since the users shouldn't have access to the logout button and it might make debugging easier? What do you think?

Reply:
Yeah, I though of the case when not-digid user can have an access to the digid logout button, but you are right, it'd better be dealt with on higher level. I'm not sure about raising an exception though, do we really want user to see 500 error?

Reply:
As discussed on Slack if non-digid user tries to logout from Digid, HTTP 403 will be returned